sklearn/decomposition/_dict_learning.py

```diff
@@ -865,6 +902,98 @@ def dict_learning_online(
     SparsePCA
     MiniBatchSparsePCA
     """
+    deps = (return_n_iter, return_inner_stats, iter_offset, inner_stats)
+    if max_iter is not None and not all(arg == "deprecated" for arg in deps):
+        raise ValueError(
+            "the following args are incompatible with 'max_iter': "
+            "return_n_iter, return_inner_stats, iter_offset, inner_stats"
+        )
+
+    if iter_offset != "deprecated":
+        warnings.warn(
+            "'iter_offset' is deprecated in version 1.1 and "
+            "will be removed in version 1.3.",
+            FutureWarning,
+        )
+    else:
+        iter_offset = 0
+
+    if return_inner_stats != "deprecated":
+        warnings.warn(
+            "'return_inner_stats' is deprecated in version 1.1 and "
+            "will be removed in version 1.3.",
+            FutureWarning,
+        )
+    else:
+        return_inner_stats = False
+
+    if inner_stats != "deprecated":
+        warnings.warn(
+            "'inner_stats' is deprecated in version 1.1 and "
+            "will be removed in version 1.3.",
+            FutureWarning,
+        )
+    else:
+        inner_stats = None
+
+    if return_n_iter != "deprecated":
+        warnings.warn(
+            "'return_n_iter' is deprecated in version 1.1 and "
+            "will be removed in version 1.3. From version 1.3 n_iter"
+            " will always be returned",
```

Comment:
I think I would rather never return it when calling the function. If someone is interested in this details, one should use the object oriented API instead I think.

Reply:
Alternatively, we can keep this option non-deprecated and only deprecate `return_inner_stats`.

Reply:
As discussed irl, let's deprecate and never return in the future